server/xls_exporter.py

```diff
@@ -0,0 +1,68 @@
+import xlwt
+import datetime
+
+import django.db.models.manager
+from django.db import models
+from django.http import HttpResponse
+
+
+def export_model_to_xls(
+        model_type,
+        fields_descriptions,
+        filename=None,
+        spreadsheet_name=None
+):
+    """
+    Exports all model objects in the database to a .xls file, and returns a response with it as an attachment.
+    Serializes:
+     - Fields with choices to their corresponding value
+     - bools to "לא" or "כן"
+     - datetime objects to a formatted time ('%Y/%m/%d %H:%M')
+     - ManyToMany relationships as a comma-joined list of Model objects str() was called on
+     - Model objects by calling str() on them
+    :param model_type: The django.db.models.Model instance to operate on
+    :param fields_descriptions: A dictionary, with keys as the field names as specified in the model, and values as
+                                the text to display in the header row
+    :param filename: The filename to generate without the .xls extension
+                     Default: <model_name_lower>_data-%Y_%m_%d-%H%M%S (with formatted time)
+    :param spreadsheet_name: The name of the spreadsheet in the file
+                             Default: <model_name>s
+
+    :return: HttpResponse of content type application/ms-excel with the .xls file as an attachment
+    """
+    response = HttpResponse(content_type='application/ms-excel')
+    response['Content-Disposition'] = 'attachment; filename="{}.xls"'.format(
+        filename if filename is not None else '{}_data-{}'.format(
+            model_type.__name__.lower(),
+            datetime.datetime.now().strftime('%Y_%m_%d-%H%M%S')
+        )
+    )
```

Comment:
instead of a one liner do: if filename is None and assign the filename as you wish

Reply:
you're right. resolved